pkg/gadgets/trace/dns/tracer/tracer.go

```diff
@@ -332,30 +132,80 @@ func (t *Tracer) install() error {
 	return nil
 }
 
-func (t *Tracer) run(ctx context.Context, logger logger.Logger, dnsTimeout time.Duration) error {
-	spec, err := loadDns()
+func (t *Tracer) parseDNSPacket(rawSample []byte, netns uint64) (*types.Event, error) {
+	// The sample received is a concatenation of the dnsEventT structure and the packet bytes.
```

Comment:
I have to be honest. I didn't understand how we are sending the skb from the eBPF code 😅.

Reply:
Answer [here](https://github.com/inspektor-gadget/inspektor-gadget/pull/2544#discussion_r1505015528).